### DISCUSSION TITLE
Use 0.0.0.0 in Axum Examples

### DISCUSSION THREAD

**Discussion by ohchase** [tokio-rs/axum#1578] [Ideas]
Should the axum examples use 0.0.0.0 as their host to map too?
When using 127.0.0.1 as the host, the axum server isn't exposable through the typical docker port attribute.
If you use 0.0.0.0:3000 and expose port 3000 through your dockerfile it works, but
If you use 127.0.0.1:3000 and expose port 3000 through your dockerfile it will not work.

Is there a reason why the examples need to use 127.0.0.1, I feel like 0.0.0.0 works locally and in docker seamlessly so might as well just use 0.0.0.0 right?

**Comment by davidpdrsn** [comment]
The examples aren't meant for docker, they're meant to show how to use axum.

With that said I'm fine with changing it. Do you wanna make a PR?

**Reply by ohchase** [reply]
Thank you for your response, I totally understand what you're saying about the examples purpose being axum and not docker; and the axum examples are undoubtedly some of the best most comprehensive examples I've seen in a repository. My suggestion stemmed from the fact, I imagined other users might have the same flow as me and run the examples and then start needing microservices or dockerfiles for container deployments so they bundle the apps. On that note, I'd be happy to make a pr

**Reply by davidpdrsn** [reply]
Thanks!

**Comment by LockedThread** [comment]
I think it might be a nice note to add

**Reply by ohchase** [reply]
Yea a note would be best I expect, the reason I never further pursued this is because I worried about the security implication of then having the sample containers by default available to the host. Also, I don't know how this behaves across operating systems.
